lib/knapsack_pro/runners/queue/base_runner.rb

```diff
@@ -59,9 +59,37 @@ def trap_signals
             Signal.trap(signal) {
               puts "#{signal} signal has been received. Terminating Knapsack Pro..."
               @@terminate_process = true
+              log_threads
             }
           end
         end
+
+        def log_threads
+          threads = Thread.list
+
+          puts
+          puts '=' * 80
+          puts "Start logging #{threads.count} detected threads."
+          puts 'Use the following backtrace to find the line of code that got stuck if the CI node hung and terminated your tests.'
+
+          threads.each do |thr|
+            puts
+            if thr == Thread.main
+              puts "Main thread and its backtrace:"
+            else
+              puts "Non-main thread inspect: #{thr.inspect}"
+              puts "Non-main thread backtrace:"
+            end
+            puts thr.backtrace.join("\n")
+            puts
+          end
```

Comment:
suggestion

Would you be willing not to abbreviate? I'd prefer to avoid abbreviations as much as possible.

```suggestion
          threads.each do |thread|
            puts
            if thread == Thread.main
              puts "Main thread and its backtrace:"
            else
              puts "Non-main thread inspect: #{thread.inspect}"
              puts "Non-main thread backtrace:"
            end
            puts thread.backtrace.join("\n")
            puts
          end
```